pTables.py

```diff
@@ -0,0 +1,235 @@
+        self.matrixTable = matrixTable
+        self.maxWidth = maxWidth
+        self.columnWidth = []
+        self.header = []
+        self.TableLength = len(self.matrixTable)
+        self.dataRowLength = 0
+        self._put_data()
+        self._get_data_row_length()
+
+
+
+    def is_table_empty(self):
+        """
+        Function checks whether the table passed to the constructor was empty or not.
+        :return: a boolean value
+        """
+        return self.TableLength == 0
+
+
+    def clear_My_table(self):
+        """
+        Clears the table of all its rows.
+        """
+        self._t.clear_rows()
+
+
+    def delete_a_row(self, row_number):
+        """
+
+        :param row_number: caller specifies the row number in which they wanted deleted from the table.
+        If the row number is valid, it is deleted from the table.
+        """
+
+        row_number = int(row_number)
+        row_number -= 1
+        if row_number >= 1 and row_number <= self.dataRowLength-1 :
+            self._t.del_row(row_number)
+
+
+    def output_table(self):
+        """
+        prints the object form of the table
+        """
+        if not self.is_table_empty():
+            print(self._t)
+
+
+    def get_string_version(self):
+        """
+        :return: The string variable of the table representation.
+        """
+        if not self.is_table_empty():
+            return self._t.get_string()
+
+
+    def sort_table(self, s):
+        """
+        function checks whether or not the string passed is in the header of the table or not. If it is not,
+        the function does nothing.
+        :param s: String passed by the caller. If not a string when classed, function converts (s) to a string.
+        :return: the string representation of the table object, sorted by the string passed by caller.
+        """
+        if not self.is_table_empty():
+            s = str(s)
+            if s in self.header:
+                return self._t.get_string(sortBy=s)
+
+
+
+########################################################################################################################
+########################################################################################################################
+############################# Private functions to class: MyTable    Please No Touch ###################################
+########################################################################################################################
+########################################################################################################################
+
+    def _get_data_row_length(self):
+        """
+        Private function to class MyTable.
+
+        Function assigns the member variable self.dataRowLength with the length of all rows in the table
+        containing the data. This length excludes the header of the table, self.matrixTable[0]
+        """
+
+        if not self.is_table_empty():
```

Comment:
If the table is empty, this function doesn't return anything. Shouldn't it return a 0, or something?

Reply:
I don't think it should. If the caller makes an instance and feeds it an empty table. This class should do nothing.